src/neo-vm/Utility.cs

```diff
@@ -25,6 +25,23 @@ static Utility()
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
 
+        public static BigInteger ModInverse(this BigInteger value, BigInteger modulus)
+        {
+            if (value <= 0) throw new ArgumentOutOfRangeException(nameof(value));
+            if (modulus < 2) throw new ArgumentOutOfRangeException(nameof(modulus));
+            BigInteger r = value, old_r = modulus, s = 1, old_s = 0;
```

Comment:
```suggestion
            BigInteger r = modulus, old_r = value, s = 0, old_s = 1;
```

This may be a matter of taste, but the sources I checked on Extended Euclidean Algorithm, the Bézout parameter `s` starts with 0 (previously 1), so as we start computing with `modulus` as `r`, being `value` the previous `r`... both work the same on practice, I guess.
Besides that, it would be just nice to put some huge number on tests, so as some negative inputs as well.

Reply:
Will there be a difference in performance?

Reply:
> Will there be a difference in performance?

A terrible question for a researcher hahaha

Consider this:
(standard) r = modulus, old_r = value, old_s = 1, s = 0 
(nonstandard) r = value, old_r = modulus, old_s = 0, s = 1

Well, I don't fully understand the pattern @erikzhang, but for the very basic examples (19,141) e (160,141):

(standard)
a=19 b=141 (6 cycles)
a=160 b=141 (6 cycles)

(nonstandard)
a=19 b=141 (5 cycles)
a=160 b=141 (7 cycles)

I imagine this is only the difference of a single flip in first iteration, but honestly, not starting with s=0 still feels like skipping one iteration to me... that's why I proposed to use that... but some people seem to use this way.

Reply:
Long logs:
```
r = modulus, old_r = value, old_s = 1, s = 0 (standard)
a=19 b=141 (6 cycles)

old_s=0	s=1	r=19	old_r=141
old_s=1	s=-7	r=8	old_r=19
old_s=-7	s=15	r=3	old_r=8
old_s=15	s=-37	r=2	old_r=3
old_s=-37	s=52	r=1	old_r=2
old_s=52	s=-141	r=0	old_r=1
old_s=52
CHECK = 1  (19 * 52 % 141)


a=160 b=141 (6 cycles)
old_s=0	s=1	r=19	old_r=141
old_s=1	s=-7	r=8	old_r=19
old_s=-7	s=15	r=3	old_r=8
old_s=15	s=-37	r=2	old_r=3
old_s=-37	s=52	r=1	old_r=2
old_s=52	s=-141	r=0	old_r=1
old_s=52
CHECK = 1  (160 * 52 % 141)


=======

r = value, old_r = modulus, old_s = 0, s = 1 (nonstandard)
a=19 b=141 (5 cycles)
old_s=1	s=-7	r=8	old_r=19
old_s=-7	s=15	r=3	old_r=8
old_s=15	s=-37	r=2	old_r=3
old_s=-37	s=52	r=1	old_r=2
old_s=52	s=-141	r=0	old_r=1
old_s=52
CHECK = 1  (19 * 52 % 141)



a=160 b=141 (7 cycles)
old_s=1	s=0	r=141	old_r=160
old_s=0	s=1	r=19	old_r=141
old_s=1	s=-7	r=8	old_r=19
old_s=-7	s=15	r=3	old_r=8
old_s=15	s=-37	r=2	old_r=3
old_s=-37	s=52	r=1	old_r=2
old_s=52	s=-141	r=0	old_r=1
old_s=52
CHECK = 1  (160 * 52 % 141)
```

Reply:
@erikzhang after some thoughts, I think it's really better to assume `s = 0` and `old_s = 1`, simply because it's standard.
On general, we can assume that we want to find:

`a*s + b*t = gcd(a,b) = 1`

meaning that, `a*s -1 = (-t)*b`,  that gives `a*s === 1 (mod b)`

So, we assume parameters ModInverse(a, b), with behavior to follow the division algorithm (considering a sequence `r`=r0, r1, r2, ...):

`r{i-2}=r{i-1}qi + ri    0 <= ri < r{i-1}`   (division algorithm)

`ri = si*a + ti*b`       (we want this)

`ri = r{i-2} - r{i-1}qi` (but we also have this)

So: 

`ri = s{i-2}a + t{i-2}b - (s{i-1}a + t{i-1}b) qi = (s{i-2} - s{i-1}qi)a + (t{i-2}-t{i-1}qi)b`

we derive:

`si = s{i-2} - s{i-1}qi`

`ti = t{i-2} - t{i-1}qi`

And finally:

`a = r0 = s0*a + t0*b => s0 = 1, t0 = 0`

`b = r1 = s1*a + t1*b => s1 = 0, t1 = 1`

In our case, we only need `s`, so we ignore Bézout parameter `t`... in any case, everything gets backwards if we assume gcd(b,a), but that's nonstandard. So, I don't think there's any performance issue, but it's good to follow the standard.

Reply:
Recursive definition also only works with order s0=1, s1=0... it seems more natural indeed.
```{c++}
auto MCDE_rec(long a, long b) {
  if (b == 0)
    return std::make_tuple(a, 1L, 0L);
  else {
    auto [old_d, old_s, old_t] = MCDE_rec(b, a % b);
    auto [d, s, t] = std::make_tuple(old_d, old_t, old_s - a / b * old_t);
    return std::make_tuple(d, s, t);
  }
}
```

Reply:
Do you agree with the parameter change @shargon ? I know your original code used s=1 s_old=0, but if you don't mind changing, I think this is truly safer.

Reply:
Optimizations could be done later if there are no denial of service, we need to do a lot of test in order to know that both methods work as expected.